engine/src/image.rs

```diff
@@ -8,18 +8,48 @@ use super::svg::SvgTangibleObject;
 pub struct Image {
     path: String,
     round: Option<usize>,
+    /// (offset(x, y), stdDeviation)
+    shadow: Option<((usize, usize), usize)>,
 }
 
 impl Image {
     pub fn new_from_path(path: String) -> Self {
-        Self { path, round: None }
+        Self {
+            path,
+            round: None,
+            shadow: None,
+        }
+    }
+
+    /// Get padding size.
+    ///
+    /// Padding size is the extra space around the image,
+    /// which is left for the shadow.
+    ///
+    /// According to gaussian blur, a pixel will be affected
+    /// by the pixels no more than (3 standard deviations + 1) px.
+    ///
+    /// This padding size will affect the size of the [`Image`].
+    pub fn get_padding(&self) -> (usize, usize) {
+        if let Some((offset, std_deviation)) = &self.shadow {
+            (
+                offset.0 + 3 * std_deviation + 1,
+                offset.1 + 3 * std_deviation + 1,
+            )
+        } else {
+            (0, 0)
+        }
     }
 }
 
 impl SizeOptionT for Image {
     fn get_size_option(&self) -> SizeOption {
```

Comment:
Need to have test coverage on expanded `get_size_option` in the future as it’s highly coupled with the `to_svg` behavior.